server/session_manager.js

```diff
@@ -0,0 +1,95 @@
+var log = require('minilog')('radar:session_manager')
+var ClientSession = require('../client/client_session')
+var EventEmitter = require('events').EventEmitter
+var inherits = require('util').inherits
+var ObservableMap = require('observable-map')
+var _ = require('underscore')
+
+function SessionManager (opt) {
+  var self = this
+  this.sessions = new ObservableMap()
+  this.sessions.on('change', function (event) {
+    self.emit('change', event)
+  })
+
+  this.adapters = opt && opt.adapters || []
+  this.adapters.forEach(function (adapter) {
```

Comment:
Yes- I'll extract https://github.com/zendesk/radar/blob/master/server/service_interface.js#L141-L161 into a ServiceInterface adapter, and any future transport layers or various client types, like if we have a separate MobileClient class, for example.